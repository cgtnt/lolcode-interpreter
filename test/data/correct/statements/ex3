HAI

I HAS A fish ITZ "bob"
I HAS A stone
I HAS A fishtwo ITZ A NUMBAR
VISIBLE fish!
stone R "two"
VISIBLE fishtwo!

KTHXBYE
