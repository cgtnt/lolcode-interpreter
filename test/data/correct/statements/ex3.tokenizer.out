@1:T_BEGIN:HAI-@3:T_COMMAND_TERMINATOR:
-@3:T_DECLARE_VAR:I HAS A-@3:T_T_IDENTIFIER:fish-@3:T_DECLARE_SET_VAR:ITZ-@3:T_T_STRING:"bob"-@4:T_COMMAND_TERMINATOR:
-@4:T_DECLARE_VAR:I HAS A-@4:T_T_IDENTIFIER:stone-@5:T_COMMAND_TERMINATOR:
-@5:T_DECLARE_VAR:I HAS A-@5:T_T_IDENTIFIER:fishtwo-@5:T_DECLARE_TYPE_VAR:ITZ A-@5:T_TI_FLOAT:NUMBAR-@6:T_COMMAND_TERMINATOR:
-@6:T_WRITE_STDOUT:VISIBLE-@6:T_T_IDENTIFIER:fish-@6:T_BANG:!-@7:T_COMMAND_TERMINATOR:
-@7:T_T_IDENTIFIER:stone-@7:T_ASSIGN:R-@7:T_T_STRING:"two"-@8:T_COMMAND_TERMINATOR:
-@8:T_WRITE_STDOUT:VISIBLE-@8:T_T_IDENTIFIER:fishtwo-@8:T_BANG:!-@10:T_COMMAND_TERMINATOR:
-@10:T_END:KTHXBYE-@11:T_COMMAND_TERMINATOR:
-@11:T_EOF:
